components/fleetArchiveDS.js

```diff
@@ -0,0 +1,187 @@
+          trip_id,
+          this.startTimeSeconds,
+          this.endTimeSeconds,
+          jwt
+        );
+        if (logs) {
+          tripLogs = _.concat(tripLogs, logs);
+        }
+      }
+    } else {
+      console.log(`no trips associated with vehicle id ${vehicle_id}`);
+    }
+    return tripLogs;
+  }
+
+  async fetchTaskLogsForVehicle(vehicle_id, vehicleLogs, jwt) {
+    if (!vehicle_id) {
+      return [];
+    }
+    console.log("Loading tasks logs for deliveryVehicle id", vehicle_id);
+    const task_ids = _(vehicleLogs)
+      .map(
+        (logEntry) =>
+          _.get(
+            logEntry,
+            "createDeliveryVehicle.response.remainingVehicleJourneySegments"
+          ) ||
+          _.get(
+            logEntry,
+            "getDeliveryVehicle.response.remainingVehicleJourneySegments"
+          ) ||
+          _.get(
+            logEntry,
+            "updateDeliveryVehicle.response.remainingVehicleJourneySegments"
+          )
+      )
+      .flatten()
+      .map((segment) => _.get(segment, "stop.tasks"))
+      .flatten()
+      .map((tasks) => _.get(tasks, "taskId"))
+      .flatten()
+      .uniq()
+      .compact()
+      .value();
+    let taskLogs = [];
+    if (task_ids.length > 20) {
+      // See https://github.com/googlemaps/fleet-debugger/issues/100
+      console.warn("Too many tasks found, limiting detailed logs to first 20");
+      task_ids = task_ids.slice(0, 20);
+    }
+    if (task_ids.length > 0) {
+      console.log("gots task_ids", task_ids);
+      for (const task_id of task_ids) {
+        const logs = await logging.fetchLogsFromArchive(
+          "tasks",
+          task_id,
+          this.startTimeSeconds,
+          this.endTimeSeconds,
+          jwt
+        );
+        if (logs) {
+          taskLogs = _.concat(taskLogs, logs);
+        }
+      }
+    } else {
+      console.log(`no tasks associated with vehicle id ${this.argv.vehicle}`);
+    }
+    return taskLogs;
+  }
+
+  async fetchVehicleLogs(vehicle, trip, jwt) {
+    const label = vehicle ? "vehicles" : "trips";
+    const labelVal = vehicle ? vehicle : trip;
+
+    console.log(`Fetching logs for ${label} = ${labelVal} from Fleet Archive`);
+    const entries = await logging.fetchLogsFromArchive(
+      label,
+      labelVal,
+      this.startTimeSeconds,
+      this.endTimeSeconds,
+      jwt
+    );
+
+    if (vehicle || entries.length === 0) return entries;
+
+    // For trip logs, need to get the logs for the vehicle and then filter on the given trip
+    console.log(`Logs found, looking for vehicle id`);
+    const vehicle_ids = _(entries)
+      .map(
+        (entry) =>
+          _.get(entry, "createTrip.response.vehicleId") ||
+          _.get(entry, "getTrip.response.vehicleId") ||
+          _.get(entry, "updateTrip.response.vehicleId")
+      )
+      .uniq()
+      .compact()
+      .value();
+    if (vehicle_ids.length === 0) return entries;
+
+    const vehicle_id = vehicle_ids[0];
```

Comment:
Check if there can be multiple vehicles for a trip

Reply:
Ok will do.

Reply:
Done